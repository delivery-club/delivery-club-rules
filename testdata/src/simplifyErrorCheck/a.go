package simplifyErrorCheck

type foo string

func a() {
	var bar foo

	err := bar.Error() // want "error check can be simplified in one line"
	if err != nil {
		return
	}

	err = bar.Error()
	if err != nil {
		return
	}

	return
}

func (foo) Error() error {
	return nil
}
