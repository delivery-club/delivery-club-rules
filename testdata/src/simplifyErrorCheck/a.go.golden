package simplifyErrorCheck

type foo string

func a() {
	var (
		bar foo
		err error
	)

	if err = bar.Error(); err != nil {
		return
	}

	if err = bar.Error(); err != nil {
		return
	}

	return
}

func b() {
	var (
		err error
		bar foo
	)

	if err = bar.Error(); err != nil {
		return
	}
}

func myFooFoo() {
	if err := foo("").Error(); err != nil {
		return
	}

	print(123)
}

func negative() {
	err, err2 := foo("").Error(), foo("").Error()
	if err != nil {
		print(123)
	}

	if err2 != nil {
		print(332)
	}
}

func (foo) Error() error {
	return nil
}
